sheets/api.py

```diff
@@ -374,25 +411,46 @@ def create_coupons_from_sheet(self):
                         date_processed=now_in_utc(),
                     )
                 )
-        return processed_requests
+        return processed_requests, failed_requests, unrecorded_complete_requests
 
     def update_coupon_request_processed_dates(self, processed_requests):
         """
-        For all processed request rows, programatically sets the "processed" column to checked/TRUE.
+        For all processed request rows, programatically sets the "Date Processed" column value and
+        blanks the error column.
 
         Args:
             processed_requests (list of ProcessedRequest): A list of ProcessedRequest objects
         """
         for processed_request in processed_requests:
-            self.coupon_request_sheet.update_value(
-                "{}{}".format(
-                    CouponRequestRow.DATE_PROCESSED_COLUMN, processed_request.row_index
-                ),
-                format_datetime_for_sheet_formula(
-                    processed_request.date_processed.astimezone(
-                        settings.SHEETS_DATE_TIMEZONE
-                    )
+            self.coupon_request_sheet.update_values(
+                crange="{processed_col}{row_index}:{error_col}{row_index}".format(
+                    processed_col=CouponRequestRow.DATE_PROCESSED_COLUMN,
+                    error_col=CouponRequestRow.ERROR_COLUMN,
+                    row_index=processed_request.row_index,
                 ),
+                values=[
+                    [
+                        format_datetime_for_sheet_formula(
+                            processed_request.date_processed.astimezone(
+                                settings.SHEETS_DATE_TIMEZONE
+                            )
+                        ),
+                        "",
+                    ]
+                ],
+            )
+
+    def update_coupon_request_errors(self, failed_requests):
+        """
+        For all failed request rows, updates the error column to contain the error message.
+
+        Args:
+            failed_requests (list of FailedRequest): A list of FailedRequest objects
+        """
+        for failed_request in failed_requests:
```

Comment:
It doesn't look like this is covered by tests. Can you add tests for this function?